### PR TITLE
Fix mplex stream close reset

### PR DESCRIPTION
**Pull request by mhchia**
Fixes https://github.com/libp2p/py-libp2p/issues/235.

### CODE REVIEW

**Comment by mhchia** [comment]
Tests fail currently. Still need debugging.

**Comment by mhchia** [comment]
Working on proper tests now.

File: libp2p/stream_muxer/mplex/mplex_stream.py [hunk 1]
```diff
@@ -90,63 +122,62 @@ def is_initiator(self) -> bool:
...
                     else HeaderTags.ResetReceiver
                 )
-                await self.mplex_conn.send_message(flag, None, self.stream_id)
-
-            self.local_closed = True
-            self.remote_closed = True
+                asyncio.ensure_future(
+                    self.mplex_conn.send_message(flag, None, self.stream_id)
+                )
+                await asyncio.sleep(0)
```
Comment (ralexstokes) [inline libp2p/stream_muxer/mplex/mplex_stream.py]: why do we need the `sleep` here?

Comment (ralexstokes) [reply libp2p/stream_muxer/mplex/mplex_stream.py]: its not clear to me how it interacts w/ the `close_lock`

Comment (ralexstokes) [reply libp2p/stream_muxer/mplex/mplex_stream.py]: yeah, i'd think the usual way to approach that would be to just `await self.mplex_conn.send_message` and then if we do need some type of synchronization we can add it, integrating the other parts of the machinery that need the `asyncio.Lock` or whatever mechanism makes sense

**Pull request merged**